### DISCUSSION TITLE
add a new objFunc

### DISCUSSION THREAD

**Discussion by DUT-wansy** [mdolab/dafoam#306] [General]
Hello Dr He,
I have already added `myobj.C` and `myobj.H` in `src/adjoint/DAObjFunc`, and add the paths to `src/adjoint/Make/files_Compressible` and `files_Incompressible`
This error occurs when I try to create a new target function. Do you know what caused this?
Thank you!
``` cpp
--> FOAM FATAL ERROR:
Unknown DAObjFunc type vorticity

Valid DAObjFunc types:

8
(
centerOfPressure
fieldInversion
force
meshQualityKS
moment
totalPressure
variableVolSum
wallHeatFlux
)


    From function DAObjFunc::New(    const fvMesh&,    const DAOption&,    const DAModel&,    const DAIndex&,    const DAResidual&,    const word,    const word,    const dictionary&)
    in file DAObjFunc/DAObjFunc.C at line 90.

FOAM exiting


```

**Comment by friedenhe** [comment]
Check https://dafoam.github.io/mydoc_get_started_faq.html#how-to-create-a-new-objective-function I added a new note which may fix the problem you had.

**Reply by DUT-wansy** [reply]
OK，Thank you，I will try it again.
